### ISSUE TITLE
Query: invalid sql (duplicate order by columns in the list) for query with select anonymous type, order by and group by the same property

### ISSUE THREAD

**Issue by maumar** [dotnet/efcore#12816]
query:

```cs
var query = ctx.OrderDetails.Select(od => new { od, customer = od.Order.CustomerID }).OrderBy(e => e.customer).GroupBy(e => e.customer, elementSelector: e => e.od).Select(e => e);
var result = query.ToList();
```

sql:

```sql
SELECT [od].[OrderID], [od].[ProductID], [od].[Discount], [od].[Quantity], [od].[UnitPrice], [od.Order].[CustomerID] AS [customer]
FROM [Order Details] AS [od]
INNER JOIN [Orders] AS [od.Order] ON [od].[OrderID] = [od.Order].[OrderID]
ORDER BY [od.Order].[CustomerID], [customer]
```

**Comment by divega** [comment]
As per the discussion in triage, removing the proposed-punt label for this. IIRC, we said this is likely going to be fixed for 3.0.

**Comment by smitpatel** [comment]
Duplicate of #17068
